com.unity.ml-agents/Runtime/Actuators/ActionSegment.cs

```diff
@@ -0,0 +1,181 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Unity.MLAgents.Actuators
+{
+    /// <summary>
+    /// ActionSegment{T} is a data structure that allows access to a segment of an underlying array
+    /// in order to avoid the copying and allocation of sub-arrays.  The segment is defined by
+    /// the offset into the original array, and an length.
+    /// </summary>
+    /// <typeparam name="T">The type of object stored in the underlying <see cref="Array"/></typeparam>
+    internal readonly struct ActionSegment<T> : IEnumerable<T>, IEquatable<ActionSegment<T>>
+        where T : struct
+    {
+        /// <summary>
+        /// The zero-based offset into the original array at which this segment starts.
+        /// </summary>
+        public readonly int Offset;
+
+        /// <summary>
+        /// The number of items this segment can access in the underlying array.
+        /// </summary>
+        public readonly int Length;
+
+        /// <summary>
+        /// An Empty segment which has an offset of 0, a Length of 0, and it's underlying array
+        /// is also empty.
+        /// </summary>
+        public static ActionSegment<T> Empty = new ActionSegment<T>(System.Array.Empty<T>(), 0, 0);
+
+#if DEBUG
```

Comment:
I think you need to either move the `#if DEBUG` into the body, or provide any empty implementation in an `#else`.

Reply:
ah, crap. nice catch